### PR TITLE
feat: Support the Count aggregation query

### PR DESCRIPTION
**Pull request by Mariatta**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-datastore/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: google/cloud/datastore/aggregation.py [hunk 1]
```diff
@@ -0,0 +1,434 @@
+# # Copyright 2014 Google LLC
```
Comment (kolea2) [inline google/cloud/datastore/aggregation.py:1]: 2022 :)

File: google/cloud/datastore/client.py [hunk 2]
```diff
@@ -837,6 +839,78 @@ def do_something_with(entity):
...
+            >>> query = client.query(kind='MyKind')
+            >>> aggregation_query.add_aggregation(CountAggregation(alias='total'))
+
+        Adding multiple aggregations to the aggregation_query
+
+        .. doctest:: aggregation_query
+
+            >>> query = client.query(kind='MyKind')
+            >>> total_count = CountAggregation(alias='total')
+            >>> count_up_to = CountAggregation(up_to=10)
```
Comment (kolea2) [inline google/cloud/datastore/client.py:880]: i don't think we are using up_to?

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: google/cloud/datastore/aggregation.py [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+        elif response_pb.batch.more_results in _FINISHED:
+            self._more_results = False
+        else:
+            raise ValueError("Unexpected value returned for `more_results`.")
+
+        return [
+            result.aggregate_properties
+            for result in response_pb.batch.aggregation_results
+        ]
+        # return query_result_from_pb(response_pb)
```
Comment (kolea2) [inline google/cloud/datastore/aggregation.py:349]: still needed?

File: tests/system/test_aggregation_query.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
+# Copyright 2011 Google LLC
```
Comment (kolea2) [inline tests/system/test_aggregation_query.py:1]: let's add some more test cases using count with an existing query (like a filter or something similar)

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: tests/system/test_aggregation_query.py (hunk 4 above)
Comment (Mariatta) [reply tests/system/test_aggregation_query.py:1]: Thanks, it's a great idea. I've added some more tests where the nested query are filtered.

File: google/cloud/datastore/aggregation.py (hunk 3 above)
Comment (Mariatta) [reply google/cloud/datastore/aggregation.py]: removed.

File: tests/system/test_aggregation_query.py (hunk 4 above)
Comment (kolea2) [reply tests/system/test_aggregation_query.py:1]: Great! I also just noticed the year on this needs updating :)

**Review by kolea2 — APPROVED** [review]
LGTM but would like someone from @googleapis/yoshi-python to also review. Thanks!!

**Review by parthea — APPROVED** [review]

File: tests/system/test_aggregation_query.py (hunk 4 above)
Comment (Mariatta) [reply tests/system/test_aggregation_query.py]: Updated.

**Pull request merged**